src/toil/test/src/promisedRequirementTest.py

```diff
@@ -123,6 +126,17 @@ def testPromisesWithJobStoreFileObjects(self):
             Job.Runner.startToil(F1, self.getOptions(self._createTempDir('testFiles')))
 
 
+        def testPromiseRequirementRace(self):
+            """
+            Checks for a race condition when using promised requirements and child job functions.
+            """
+            A = Job.wrapJobFn(logDiskUsage, 'A', sleep=5, disk=PromisedRequirement(1024))
```

Comment:
It shouldn't.  You can pass a single value or promise and it will just return that value.